test/integrated/pond_soil_lsm.jl

```diff
@@ -113,25 +113,23 @@ for FT in (Float32, Float64)
         land_auxiliary_vars = auxiliary_vars(land)
         land_auxiliary_types = auxiliary_types(land)
         # prognostic vars
-        @test land_prognostic_vars.soil == prognostic_vars(land.soil)
-        @test land_prognostic_vars.surface_water ==
+        @test land_prognostic_vars[1].soil == prognostic_vars(land.soil)
```

Comment:
why do we index into the vars with integer indices? I don't understand the structure of prognostic and aux vars here

Reply:
yeah, Im not sure what to do here. Before, we returned a NamedTuple, for prognostic and aux vars, so we could access them like: `land_prognostic_vars.soil`, without the indexing. This is slightly inconsistent between integrated and standalone models, since `prognostic_vars(standalone model)` returns a tuple (e.g. `prognostic_vars(::RichardsModel) = (:\vartheta_l,)`. But for integrated models, we would make use of the names and return `(:soil = (:\vartheta_l,), :canopy = (...))`.

The additional aux variables were stored in the NamedTuple with the key `interactions`. I wasnt sure that was clear anymore, but without a key, returning a Tuple felt like it made the most sense. And if I did it for aux, I felt like it was good to be consistent with prognostic. but then we get the indexing.

Maybe I can change back to how it was (NamedTuples), and just replace the key `:interactions` with `lsm_aux` ?

Reply:
I think I will do that! it will result in a smaller change in the PR compared with main, and we can do a one off PR later to address it if we feel like we need to. when we add forcing variables, it may change a little also.